internal/admin/album_coverage.go

```
@@ -0,0 +1,156 @@
+		logger.Warn("player: resolve upscale target for album coverage", "err", err)
+		return nil
+	}
+	epoch := s.catalogEpoch.Load()
+	if c := s.coverage.Load(); c != nil && c.epoch == epoch &&
+		c.rate == rate && c.bits == bits && time.Since(c.builtAt) < coverageTTL {
+		return c.byAlbum
+	}
+
+	// Same singleflight discipline as the catalog: N tabs hitting an
+	// expired snapshot at once collapse to one rebuild.
+	v, err, _ := s.coverageSF.Do("coverage", func() (any, error) {
```

golangci-lint (2.12.2)

[error] Function `albumCoverageFor$1` should pass the context parameter

(contextcheck)